tests/reader_writer_100hz.test

```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0"?>
+<launch>
+  <!-- Start Kvaser Can Bridge with Read/Write Ticks Enabled -->
+  <node name="kvaser_can_bridge" pkg="kvaser_interface" type="kvaser_can_bridge" ns="reader">
+    <param name="can_hardware_id" value="1" />
+    <param name="can_circuit_id" value="0" />
+    <param name="can_bit_rate" value="500000" />
+    <param name="enable_debug_ticks" value="true" />
+  </node>
+
+  <!-- Start Kvaser Can Bridge with Read/Write Ticks Enabled -->
+  <node name="kvaser_can_bridge" pkg="kvaser_interface" type="kvaser_can_bridge" ns="writer">
+    <param name="can_hardware_id" value="1" />
+    <param name="can_circuit_id" value="1" />
+    <param name="can_bit_rate" value="500000" />
+    <param name="enable_debug_ticks" value="true" />
+  </node>
+
+  <node name="topic_pub" type="rostopic" pkg="rostopic"
+    args="pub -r 100 /writer/can_rx can_msgs/Frame '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: ''}, id: 546, is_rtr: false, is_extended: false, is_error: false, dlc: 8, data: [0,0,0,0,0,0,0,0]}'" />
+
+  <test test-name="bridge_100hz_write_tick" pkg="rostest" type="hztest" time-limit="10.0">
+    <param name="hz" value="100.0" />
+    <param name="hzerror" value="3.0" />
+    <param name="test_duration" value="3.0" />
+    <param name="topic" value="/writer/can_write_tick" />
+  </test>
+
+  <test test-name="bridge_100hz_read_tick" pkg="rostest" type="hztest" time-limit="10.0">
+    <param name="hz" value="100.0" />
+    <param name="hzerror" value="3.0" />
+    <param name="test_duration" value="3.0" />
+    <param name="topic" value="/reader/can_read_tick" />
+  </test>
+
+  <test test-name="bridge_100hz_pub" pkg="rostest" type="hztest" time-limit="10.0">
+    <param name="hz" value="100.0" />
+    <param name="hzerror" value="3.0" />
+    <param name="test_duration" value="3.0" />
+    <param name="topic" value="/reader/can_tx" />
+  </test>
```

Comment:
Since we're checking the rate of the reader/can_tx topic directly here, does that make the /reader/can_read tick rate test redundant?

Reply:
The ticks are intended as an indication that the underlying hardware read/write mechanism completed successfully. The order of operations for this test looks like this:

- Receive CAN message on /writer/can_rx
- Message written to hardware in writer node
- if (previous step == success) then publish to /writer/can_write_tick
- Message read from hardware in reader node
- if (previous step == success) then publish to /reader/can_read_tick
- Publish CAN message to /reader/can_tx

This verifies that the ROS layer _and_ the `kvaser_interface` layer are working.

Reply:
but if the receiving `kvaser_interface` node is publishing at the correct rate, doesn't that prove that the underlying hardware is working? If messages aren't getting through correctly then we'd get error frames or bad content at the end

Reply:
That's true but the inverse is not. Just because the receiving node _isn't_ publishing doesn't mean that it isn't receiving messages at the `kvaser_interface` layer.

Reply:
I think that the lower-level test of that (does the actual `can_rx_callback` function of the rx node behave as expected) would be better handled in a unit test.

At the node level here it seems like a good acceptance criterium would be if we get back out the same messages that we put in.

Reply:
Yeah, you're probably right. I can refactor that part and remove the ticks.